crates/uv/src/main.rs

```diff
@@ -384,6 +384,16 @@ struct PipCompileArgs {
     #[clap(long, conflicts_with = "index_url", conflicts_with = "extra_index_url")]
     no_index: bool,
 
+    /// When resolving against multiple index URLs, search for every package on every index, and
+    /// merge the results.
+    ///
+    /// By default, `uv` will stop at the first index on which a given package is available, and
+    /// limit resolutions to those present on that first index. This prevents "dependency confusion"
+    /// attacks, whereby an attack can upload a malicious package under the same name to a secondary
+    /// index.
+    #[clap(long)]
+    unsafe_index_merge: bool,
```

Comment:
Should we use a enum instead e.g. `--unsafe-index-strategy flatten` or `--index-strategy unsafe-merge`? so this is future-proof?

Reply:
Yeah good call.